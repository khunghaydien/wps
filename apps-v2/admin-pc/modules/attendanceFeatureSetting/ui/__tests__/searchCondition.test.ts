import snapshotDiff from 'snapshot-diff';

import reducer, {
  ACTION_TYPE,
  actions,
  initialState,
} from '../searchCondition';

describe('reducer()', () => {
  it('@@init', () => {
    const next = reducer(undefined, { type: '@@INIT' });
    expect(snapshotDiff({}, next)).toMatchSnapshot();
  });
  it(ACTION_TYPE.INIT, () => {
    const state = {
      targetDate: '2019-10-01',
    };
    const next = reducer(state, actions.initialize());
    expect(snapshotDiff(state, next)).toMatchSnapshot();
  });
  it(ACTION_TYPE.SET, () => {
    const next = reducer(initialState, actions.set('targetDate', 'xxx'));
    expect(snapshotDiff(initialState, next)).toMatchSnapshot();
  });
});
