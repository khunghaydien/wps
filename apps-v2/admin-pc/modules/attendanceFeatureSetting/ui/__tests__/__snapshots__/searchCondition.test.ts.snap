// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer() @@init 1`] = `
"Snapshot Diff:
- First value
+ Second value

- Object {}
+ Object {
+   \\"targetDate\\": \\"\\",
+ }"
`;

exports[`reducer() ADMIN-PC/MODULES/FEATURE-SETTING/UI/SEARCH_CONDITION/INITIALIZE 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"targetDate\\": \\"2019-10-01\\",
+   \\"targetDate\\": \\"\\",
  }"
`;

exports[`reducer() ADMIN-PC/MODULES/FEATURE-SETTING/UI/SEARCH_CONDITION/SET 1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
-   \\"targetDate\\": \\"\\",
+   \\"targetDate\\": \\"xxx\\",
  }"
`;
